torch/utils/data/_utils/pin_memory.py

```diff
@@ -35,23 +35,23 @@ def _pin_memory_loop(in_queue, out_queue, device_id, done_event):
         elif isinstance(r[1], ExceptionWrapper):
             out_queue.put(r)
         else:
-            idx, batch = r
+            idx, data = r
             try:
-                batch = pin_memory_batch(batch)
+                data = pin_memory_data(data)
             except Exception:
                 out_queue.put((idx, ExceptionWrapper(sys.exc_info())))
             else:
-                out_queue.put((idx, batch))
+                out_queue.put((idx, data))
 
 
-def pin_memory_batch(batch):
-    if isinstance(batch, torch.Tensor):
-        return batch.pin_memory()
-    elif isinstance(batch, string_classes):
-        return batch
-    elif isinstance(batch, container_abcs.Mapping):
-        return {k: pin_memory_batch(sample) for k, sample in batch.items()}
-    elif isinstance(batch, container_abcs.Sequence):
-        return [pin_memory_batch(sample) for sample in batch]
+def pin_memory_data(data):
+    if isinstance(data, torch.Tensor):
+        return data.pin_memory()
+    elif isinstance(data, string_classes):
+        return data
+    elif isinstance(data, container_abcs.Mapping):
+        return {k: pin_memory_data(sample) for k, sample in data.items()}
+    elif isinstance(data, container_abcs.Sequence):
+        return [pin_memory_data(sample) for sample in data]
     else:
-        return batch
+        return data
```

Comment:
Because data loader can be used to load individual samples as well, and this PR provides better support for that use case. `batch` seems to assume that the data is always batched.